src/Agent/NewRelic/Agent/Extensions/Providers/Wrapper/Kafka/KafkaConsumerWrapper.cs

```diff
@@ -80,15 +88,69 @@ public AfterWrappedMethodDelegate BeforeWrappedMethod(InstrumentedMethodCall ins
             });
         }
 
+        private static long GetHeadersSize(Headers headers)
+        {
+            var headersSize = 0L;
+            if (headers != null)
+            {
+                foreach (var header in headers)
+                {
+                    headersSize += Encoding.UTF8.GetByteCount(header.Key);
+                    headersSize += header.GetValueBytes().Length;
+                }
+            }
+            return headersSize;
+        }
+
+        private static void ReportSizeMetrics(IAgent agent, ITransaction transaction, string topic, long headersSize, object messageAsObject)
+        {
+            // get the message Key and Value properties so we can try to get their size
+            var messageType = messageAsObject.GetType();
+            var keyAccessor = KeyAccessorDictionary.GetOrAdd(messageType, GetKeyAccessorFunc);
+            var valueAccessor = ValueAccessorDictionary.GetOrAdd(messageType, GetValueAccessorFunc);
+
+            var keyAsObject = keyAccessor(messageAsObject);
+            var valueAsObject = valueAccessor(messageAsObject);
+
+            var totalSize = headersSize + TryGetSize(keyAsObject) + TryGetSize(valueAsObject);
+
+            if (totalSize > 0)
+            {
+                transaction.AddCustomAttribute("kafka.consume.byteCount", totalSize);
+            }
+
+            // Add metrics for bytes received and messages received
+            var agentExp = agent.GetExperimentalApi();
+            agentExp.RecordCountMetric($"Message/Kafka/Topic/Named/{topic}/Received/Messages", 1);
+            agentExp.RecordCountMetric($"Message/Kafka/Topic/Named/{topic}/Received/Bytes", totalSize);
```

Comment:
I think that this Bytes metric should not be a Count metric, because it will only report that bytes value in the count field, which will prevent the UI from performing useful functions on it. This should probably be reported as a summary metric (which will allow usage of all of the functions in the UI).

Reply:
@nrcventura I'd be happy to make that change if I understood a bit more about how metric payloads work. I'll dig around and see what I can figure out and then ask for help if I get stuck... ;-)